libs/blocks/modal/modal.js

```diff
@@ -26,7 +26,12 @@ function getDetails(el) {
 function closeModals(modals) {
   const qModals = modals || document.querySelectorAll('dialog[open]');
   if (qModals?.length) {
-    qModals.forEach((modal) => { modal.remove(); });
+    qModals.forEach((modal) => {
+      if (modal.nextElementSibling.classList.contains('modal-curtain')) {
```

Comment:
maybe harden just in case `nextElementSibling` is null?
```suggestion
      if (modal.nextElementSibling?.classList.contains('modal-curtain')) {
```

Reply:
Done.